src/include/txn.i

```diff
@@ -444,7 +444,8 @@ __wt_txn_pinned_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *pinned_tsp)
 	    __wt_timestamp_set(&checkpoint_ts,
 	    &txn_global->checkpoint_timestamp));
 
-	if (__wt_timestamp_cmp(&checkpoint_ts, &pinned_ts))
+	if (!__wt_timestamp_iszero(&checkpoint_ts) &&
+	    __wt_timestamp_cmp(&checkpoint_ts, &pinned_ts))
```

Comment:
The intent is to make this match the transaction ID semantic here:
https://github.com/wiredtiger/wiredtiger/blob/develop/src/include/txn.i#L391

Reply:
I think it is better to use '<= 0' with __wt_timestamp_cmp function as it can return "-1, 0, 1".